R/FilteredData.R

```diff
@@ -155,6 +155,16 @@ FilteredData <- R6::R6Class( # nolint
       invisible(NULL)
     },
 
+    #' Get list of filter states available for this object.
```

Comment:
https://github.com/insightsengineering/teal/blob/867_mapping_matrix_NA%40main/R/module_filter_manager.R#L208
`set_available_teal_slices` filters only available datanames - but should it also filter by `attr(self$get_filter_state(), "include_varnames")`?

Reply:
Good question. 
`FilterStates` will not allow such a state to be created from the UI but what if a `teal_slices` is passed that refers to forbidden variables? This is why #340 would be nice to have.

Let me run some tests. I will modify `$set_available_teal_slices` if necessary.

Reply:
Yup, #340 is nice to have but MAE make `teal_slice` objects hard to match with include/exclude_varnames without introducing MAE-specific-code (against open-close principle).

Reply:
`teal_slice_expr` will be included if `dataname` checks out, even if they refer to variables not allowed.